pkg/mcastmanager/mcastmanager.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mcastmanager
+
+import (
+	"github.com/cilium/cilium/pkg/addressing"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/multicast"
+)
+
+var (
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "mcast-manager")
+)
+
+// TODO(ygui): Garbage collection
+
+// MCastManager manages IPv6 address
+type MCastManager struct {
+	mutex lock.Mutex
+
+	// ifc is the interface that mcast addresses are applied to
+	ifc string
+
+	// ipv6s keeps track of all ipv6 addresses by grouping them based on the
+	// last 3 bytes of the address
+	ipv6s map[int32]map[string]bool
```

Comment:
It doesn't look like `bool` here is used. Couldn't we make this into a map of `int32 -> []string`?

Reply:
It's for easy lookup, maybe `map[string]struct{}`?